src/Nixfmt/Types.hs

```diff
@@ -0,0 +1,242 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Nixfmt.Types where
+
+import           Prelude         hiding (String)
+
+import           Data.Text       hiding (concat, map)
+import           Data.Void
+import           Text.Megaparsec (Parsec)
+
+type Parser = Parsec Void Text
+
+data Trivium = EmptyLine
+             | LineComment     Text
+             | BlockComment    [Text]
+
+type Trivia = [Trivium]
+
+instance Show Trivium where
+    show EmptyLine         = "EmptyLine"
+    show (LineComment lc)  = " # " <> unpack lc
+    show (BlockComment bc) = "/*" <> concat (map unpack bc) <> "*/"
```

Comment:
Oh, that’s quite a bad idea, `Show` is usually used only for debugging output, not for actual user-facing output. I’m not sure if the formatting library that you use provides some sort of typeclass for this, but if it doesn’t I personally prefer the `fmt` library and its `Buildable` class, that allows you to work directly with `Text` without all these conversions that you have to deal with. And it’s just better semantically.